lsp/ARCHITECTURE.md

```diff
@@ -0,0 +1,110 @@
+# Architechture
+
+## Language Service
+
+The following is a class diagram for the underlying Language Service.
+These are the components that allow us to provide a text document and get
```

Comment:
Can you be more specific around what entity is "us" - does this mean language server to language service, Toolkit/IDE to language server, ...

Reply:
Updated to add more clarity